shared-bindings/bleio/Peripheral.c

```diff
@@ -0,0 +1,250 @@
+        service_ptr->device = MP_OBJ_FROM_PTR(self);
+        mp_obj_list_append(self->service_list, service);
+    }
+
+    const mp_obj_t name = args[ARG_name].u_obj;
+    if (name == mp_const_none) {
+        self->name = mp_obj_new_str(default_name, strlen(default_name));
+    } else if (MP_OBJ_IS_STR(name)) {
+        self->name = name;
+    } else {
+        mp_raise_ValueError(translate("name must be a string"));
+    }
+
+    // Do port-specific initialization.
+    common_hal_bleio_peripheral_construct(self);
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|   .. attribute:: connected
+//|
+//|     True if connected to a BLE Central device.
+//|
+STATIC mp_obj_t bleio_peripheral_get_connected(mp_obj_t self_in) {
+    bleio_peripheral_obj_t *self = MP_OBJ_TO_PTR(self_in);
+
+    // Return list as a tuple so user won't be able to change it.
+    return mp_obj_new_bool(common_hal_bleio_peripheral_get_connected(self));
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(bleio_peripheral_get_connected_obj, bleio_peripheral_get_connected);
+
+const mp_obj_property_t bleio_peripheral_connected_obj = {
+    .base.type = &mp_type_property,
+    .proxy = { (mp_obj_t)&bleio_peripheral_get_connected_obj,
+               (mp_obj_t)&mp_const_none_obj,
+               (mp_obj_t)&mp_const_none_obj },
+};
+
+//|   .. attribute:: services
+//|
+//|     A `tuple` of `bleio.Service` that are offered by this peripheral. (read-only)
+//|
+STATIC mp_obj_t bleio_peripheral_get_services(mp_obj_t self_in) {
+    bleio_peripheral_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    // Return list as a tuple so user won't be able to change it.
+    mp_obj_list_t *service_list = MP_OBJ_TO_PTR(self->service_list);
+    return mp_obj_new_tuple(service_list->len, service_list->items);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(bleio_peripheral_get_services_obj, bleio_peripheral_get_services);
+
+const mp_obj_property_t bleio_peripheral_services_obj = {
+    .base.type = &mp_type_property,
+    .proxy = { (mp_obj_t)&bleio_peripheral_get_services_obj,
+               (mp_obj_t)&mp_const_none_obj,
+               (mp_obj_t)&mp_const_none_obj },
+};
+
+//|   .. attribute:: name
+//|
+//|     The peripheral's name, included when advertising. (read-only)
+//|
+STATIC mp_obj_t bleio_peripheral_get_name(mp_obj_t self_in) {
+    bleio_peripheral_obj_t *self = MP_OBJ_TO_PTR(self_in);
+
+    return self->name;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(bleio_peripheral_get_name_obj, bleio_peripheral_get_name);
+
+const mp_obj_property_t bleio_peripheral_name_obj = {
+    .base.type = &mp_type_property,
+    .proxy = { (mp_obj_t)&bleio_peripheral_get_name_obj,
+               (mp_obj_t)&mp_const_none_obj,
+               (mp_obj_t)&mp_const_none_obj },
+};
+
+//|   .. method:: start_advertising(*, connectable=True, data=None)
+//|
+//|     Starts advertising the peripheral. The peripheral's name and
+//|     services are included in the advertisement packets.
+//|
```

Comment:
Please add more detail about what happens when data is None. Thanks for making them kw-only args.